src/vgg.jl

```diff
@@ -0,0 +1,248 @@
+- `dropout`: the dropout level between each fully connected layer
+"""
+function vgg_classifier_layers(imsize, nclasses, fcsize, dropout)
+  layers = []
+  push!(layers, flatten)
+  push!(layers, Dense(Int(prod(imsize)), fcsize, relu))
+  push!(layers, Dropout(dropout))
+  push!(layers, Dense(fcsize, fcsize, relu))
+  push!(layers, Dropout(dropout))
+  push!(layers, Dense(fcsize, nclasses))
+
+  return layers
+end
+
+"""
+    vgg(imsize; config, inchannels, batchnorm=false, nclasses, fcsize, dropout)
+
+Create a VGG model
+([reference](https://arxiv.org/abs/1409.1556v6)).
+
+# Arguments
+- `imsize`: input image width and height as a tuple
+- `config`: the configuration for the convolution layers
+            (see [`Metalhead.vgg_convolutional_layers`](#))
+- `inchannels`: number of input channels
+- `batchnorm`: set to `true` to use batch normalization after each convolution
+- `nclasses`: number of output classes
+- `fcsize`: intermediate fully connected layer size
+            (see [`Metalhead.vgg_classifier_layers`](#))
+- `dropout`: dropout level between fully connected layers
+"""
+function vgg(imsize; config, inchannels, batchnorm=false, nclasses, fcsize, dropout)
+  conv = vgg_convolutional_layers(config, batchnorm, inchannels)
+  imsize = outputsize(conv, (imsize..., inchannels); padbatch=true)[1:3]
+  class = vgg_classifier_layers(imsize, nclasses, fcsize, dropout)
+  return Chain(conv..., class...)
+end
+
+const vgg_config = Dict(:A => [(64,1), (128,1), (256,2), (512,2), (512,2)],
+                        :B => [(64,2), (128,2), (256,2), (512,2), (512,2)],
+                        :D => [(64,2), (128,2), (256,3), (512,3), (512,3)],
+                        :E => [(64,2), (128,2), (256,4), (512,4), (512,4)])
+
+"""
+    VGG{BN}(imsize=(224, 224); config, inchannels, nclasses, fcsize, dropout)
+
+Create a `VGG` model
+([reference](https://arxiv.org/abs/1409.1556v6)).
+See also [`vgg`](#).
+
+Set `BN` to `true` to enable batch normalization after
+convolution layers. Defaults to `false`.
+
+# Arguments
+- `imsize`: input image width and height as a tuple
+- `config`: the configuration for the convolution layers
+            (see [`Metalhead.vgg_convolutional_layers`](#))
+- `inchannels`: number of input channels
+- `nclasses`: number of output classes
+- `fcsize`: intermediate fully connected layer size
+            (see [`Metalhead.vgg_classifier_layers`](#))
+- `dropout`: dropout level between fully connected layers
+"""
+struct VGG{BN, T}
+  layers::T
+end
+
+function VGG{BN}(imsize=(224, 224); config, inchannels, nclasses, fcsize, dropout) where BN
+  layers = vgg(imsize; config=config,
+                        inchannels=inchannels,
+                        nclasses=nclasses,
+                        fcsize=fcsize,
+                        dropout=dropout,
+                        batchnorm=BN)
+  
+  VGG{BN, typeof(layers)}(layers)
+end
+
+Functors.functor(::Type{VGG{BN, T}}, m) where {BN, T} =
```

Comment:
Do not overload it directly. Why aren't we using `@functor` here?

Reply:
Because we need to close over the type parameters in `re`. FWIW, this is [the recommended way](https://github.com/FluxML/Functors.jl/issues/3#issuecomment-626747663) of dealing with this according to Mike.

Reply:
I am confused by why VGG needs a type param for batchnorm/no batchrnorm. Unlike ResNet where you'd potentially want to dispatch on the type of block, how many scenarios are there where you'd want code to behave differently depending on whether a VGG model has batchnorm?

Reply:
My thinking was to try and make the type be more than a plain wrapper, but maybe it just isn't worth it.

Reply:
So I think now I am going to forego the type parameters in `ResNet` and `VGG` since we can't provide similar types across all models, and I don't think they are that useful. In `ResNet`, it would be nice to distinguish basic block from bottleneck, but the current approach seems too hacky to me.